googleapiclient/discovery.py

```diff
@@ -718,7 +719,11 @@ def method(self, **kwargs):
 
     for name in parameters.required_params:
       if name not in kwargs:
-        raise TypeError('Missing required parameter "%s"' % name)
+        # temporary workaround for non-paging methods incorrectly requiring
+        # page token parameter (cf. drive.changes.watch vs. drive.changes.list)
+        if name not in _PAGE_TOKEN_NAMES or _findPageTokenName(
```

Comment:
What is the ultimate goal here? This method needs to _not_ require a `pageToken` or `nextPageToken`?

I am really struggling to follow this `if` block:

  * If `pageToken` is set as a required parameter and not provided, then the first disjunct will fail, and then `_findPageTokenName` will return a string, which is truthy (even if the string is `nextPageToken`), and we get an error. That may or may not be correct behavior (and sure does not seem like it is correct behavior on purpose).
  * If `foo` is a required parameter, then it is not in `_PAGE_TOKEN_NAMES`, and the first disjunct is true, and you get the exception. Okay, fine.